packages/flutter/lib/src/gestures/monodrag.dart

```diff
@@ -162,20 +178,31 @@ abstract class DragGestureRecognizer extends OneSequenceGestureRecognizer {
     if (_state != _DragState.accepted) {
       _state = _DragState.accepted;
       final Offset delta = _pendingDragOffset;
+      Offset updateDelta = delta;
       final Duration timestamp = _lastPendingEventTimestamp;
+      Offset startPosition;
+      switch (dragStartBehavior) {
+        case DragStartBehavior.start:
+          startPosition = _initialPosition + delta;
+          updateDelta = Offset.zero;
+          break;
+        case DragStartBehavior.down:
+          startPosition = _initialPosition;
+          break;
+      }
       _pendingDragOffset = Offset.zero;
       _lastPendingEventTimestamp = null;
       if (onStart != null) {
         invokeCallback<void>('onStart', () => onStart(new DragStartDetails(
           sourceTimeStamp: timestamp,
-          globalPosition: _initialPosition,
+          globalPosition: startPosition,
         )));
       }
-      if (delta != Offset.zero && onUpdate != null) {
+      if (updateDelta != Offset.zero && onUpdate != null) {
         invokeCallback<void>('onUpdate', () => onUpdate(new DragUpdateDetails(
           sourceTimeStamp: timestamp,
-          delta: _getDeltaForDetails(delta),
-          primaryDelta: _getPrimaryValueFromOffset(delta),
+          delta: _getDeltaForDetails(updateDelta),
```

Comment:
i'm confused. It seems like if dragStartBehavior is DragStartBehavior.start, the updateDelta will always be zero?

Reply:
Yes. Because in DragStartBehavior.start, the drag event is considered to begin at the current position (as of when onStart is called).

So `startPosition` consumes any delta that may exist at that moment, after which there is no reason to call onUpdate.